core/app/models/spree/product.rb

```diff
@@ -105,7 +105,7 @@ class Product < Spree::Base
       validates :price, if: proc { Spree::Config[:require_master_price] }
     end
 
-    validates :slug, length: { minimum: 3 }, allow_blank: true, uniqueness: true
+    validates :slug, allow_blank: true, uniqueness: true
```

Comment:
@krzysiek1507 I think `allow_blank` comes from validating uniqueness. I'm not sure whether we'll have any issues storing the records if we remove `allow_blank: true`. I didn't want to touch it as it was there from before and wasn't affecting me.